### PR TITLE
add set codecs

### PR DESCRIPTION
**Pull request by SaitTalhaNisanci**
adds set codecs, #152.

### CODE REVIEW

File: internal/protocol/set_addall.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	. "github.com/hazelcast/hazelcast-go-client/internal/serialization"
+
+	. "github.com/hazelcast/hazelcast-go-client/internal/common"
+)
+
+type SetAddAllResponseParameters struct {
+	Response bool
+}
+
+func SetAddAllCalculateSize(name *string, valueList []*Data) int {
```
Comment (furkansenharputlu) [inline internal/protocol/set_addall.go]: This should not be pointer array. Like this `[]Data`.

File: internal/protocol/set_addall.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	dataSize := 0
+	dataSize += StringCalculateSize(name)
+	dataSize += INT_SIZE_IN_BYTES
+	for _, valueListItem := range valueList {
+		dataSize += DataCalculateSize(valueListItem)
+	}
+	return dataSize
+}
+
+func SetAddAllEncodeRequest(name *string, valueList []*Data) *ClientMessage {
```
Comment (furkansenharputlu) [inline internal/protocol/set_addall.go]: Here, also. There are many `[]*Data`s to be changed to `[]Data`.

**Comment by SaitTalhaNisanci** [comment]
@furkansenharputlu 
For codecs we are using flyweight model, so for example DataCalculateSize takes a pointer and if we dont use `[]*Data` we would be copying the content of Data when calculating the size which is not necessary i.e. `DataCalculateSize(*data)`. Can you explain why you think it should be the other way ?

**Comment by SaitTalhaNisanci** [comment]
For example https://github.com/hazelcast/hazelcast-go-client/pull/154/files#diff-421d1d29fb9bb75de9fba568f5d9203bR49 here 'ReadData()' returns a pointer and if we didnt make it a pointer slice we would be copying the content of Data and taking the address afterwards, am I missing something ?

**Comment by furkansenharputlu** [comment]
When I run this code:
```package main

import "fmt"

type student struct {
	name string
}

func main() {
	var array []student = []student{student{"furkan"}, student{"talha"}}
	x(array)
	fmt.Println(array)
}

func x(pass []student) {
	pass[1].name = "talha nişancı"
}

```

The result is:
```[{furkan} {talha nişancı}]```

This is why I am saying we should use `[]Data`. I think passed slice is not deeply copied.

**Comment by SaitTalhaNisanci** [comment]
Your example is about slices so. Everything is pass by value in go except slices and maps, your example is about `*[]Data` vs `[]Data` not about `[]*Data` vs `[]Data`.  I am talking about this https://play.golang.org/p/rFJpczSd6Hn

**Comment by furkansenharputlu** [comment]
I understand your reasoning. We both think same about slices. They are not deeply copied while passing as parameter. However, you try to keep consistency in codecs so you use slice of pointers. 
Ok 👍

**Pull request merged**